### PR TITLE
azcontext: mount azure.json and file out azcontext

### PR DESCRIPTION
**Pull request by akshaysngupta**
This PR adds the code to mount the azure.json file and fill out the az context data struct.

This context will provide us with credentials and vnet details.
Following PR will be about using this.

Sample azure.json
```json
{
    "cloud": "xxxx",
    "tenantId": "t",
    "subscriptionId": "s",
    "aadClientId": "c",
    "aadClientSecret": "cs",
    "resourceGroup": "r",
    "location": "l",
    "vmType": "xxxx",
    "subnetName": "xxxx",
    "securityGroupName": "xxxx",
    "vnetName": "xxxx",
    "vnetResourceGroup": "xxxx",
    "routeTableName": "xxxx",
    "primaryAvailabilitySetName": "xxxx",
    "primaryScaleSetName": "xxxx",
    "cloudProviderBackoff": "xxxx",
    "cloudProviderBackoffRetries": "xxxx",
    "cloudProviderBackoffExponent": "xxxx",
    "cloudProviderBackoffDuration": "xxxx",
    "cloudProviderBackoffJitter": "xxxx",
    "cloudProviderRatelimit": "xxxx",
    "cloudProviderRateLimitQPS": "xxxx",
    "cloudProviderRateLimitBucket": "xxxx",
    "useManagedIdentityExtension": "xxxx",
    "userAssignedIdentityID": "xxxx",
    "useInstanceMetadata": true,
    "loadBalancerSku": "xxxx",
    "excludeMasterFromStandardLB": "xxxx",
    "providerVaultName": "xxxx",
    "maximumLoadBalancerRuleCount": "xxxx",
    "providerKeyName": "xxxx",
    "providerKeyVersion": "xxxx"
}
```

### CODE REVIEW

File: cmd/appgw-ingress/main.go [hunk 1]
```diff
@@ -88,15 +88,22 @@ func main() {
...
 	if env.AppGwName == "" {
 		env.AppGwName = env.ReleaseName
 	}
 
-	if infraSubID, infraResourceGp, err := k8sContext.GetInfrastructureResourceGroupID(); env.SubscriptionID == "" && err != nil {
-		env.SubscriptionID = string(infraSubID)
-		if env.ResourceGroupName == "" {
-			env.ResourceGroupName = string(infraResourceGp)
-		}
+	if azContext != nil && env.SubscriptionID == "" {
```
Comment (draychev) [inline cmd/appgw-ingress/main.go]: It seems that we are providing the option to read Subscription from an environment variable, which could overwrite the local file?
This looks good. If we don't have a real scenario, which requires that - we could even remove the option to provide Subscr ID via an env var to simplify.

Comment (akshaysngupta) [reply cmd/appgw-ingress/main.go]: yes, we can think about this.

Comment (akshaysngupta) [reply cmd/appgw-ingress/main.go]: We may have to think about bare metal

**Comment by akshaysngupta** [comment]
yes, will check this.

**Pull request merged**